test/integration/node-specific/resource_tracking_script_builder.ts

```diff
@@ -203,18 +208,23 @@ export async function runScriptAndGetProcessInfo(
 
   // assertions about exit status
   if (exitCode) {
+    const { error } = messages.find(m => m.error != null);
+    expect(error).to.exist;
```

Comment:
```suggestion
    expect(error, 'test script exited with non-zero exit code but did not report an error.').to.exist;
```
(optional) lately I've been adding assertion messages to preconditions like this, so that we can more easily understand exactly what failed when we see errors in our test output.

Reply:
sure. makes sense